### PR TITLE
Feat/unit test

### PR DESCRIPTION
**Pull request by sidharthjain-zemoso**

### CODE REVIEW

File: checkr/test/middlewares/auth.ts
Comment (akshayvijayjain) [inline checkr/test/middlewares/auth.ts:1]: please name test files, as ending with .spec.ts or .test.ts

File: checkr/src/services/candidate.ts [hunk 1]
```diff
@@ -60,15 +60,24 @@ const candidateService: CandidateService = {
             return { data: mappedCandidates, totalCount };
         } catch (error) {
             console.log(error);
+            if (error instanceof CustomError) {
```
Comment (akshayvijayjain) [inline checkr/src/services/candidate.ts]: think how you can move this logic to CustomError Class

**Review by akshayvijayjain — CHANGES_REQUESTED** [review]

File: checkr/src/services/candidate.ts (hunk 1 above)
Comment (sidharthjain-zemoso) [reply checkr/src/services/candidate.ts]: Here, we are throwing customError with some message in try block and in catch block we earlier were throwing again a custom error with another message which overrides the previous custom error message. To handle that we have this logic. 
Now, to move this logic to Custom Error class, should we pass error as constructor argument and then check if it is instance of custom error @akshayvijayjain ?

File: checkr/test/middlewares/auth.ts
Comment (sidharthjain-zemoso) [reply checkr/test/middlewares/auth.ts:1]: Done

File: checkr/src/common/interfaces/custom-error.ts [hunk 2]
```diff
@@ -1,9 +1,14 @@
 class CustomError extends Error {
     statusCode: number;
 
-    constructor(message: string, statusCode: number) {
-        super(message);
-        this.statusCode = statusCode;
+    constructor(message: string, statusCode: number, error?: any) {
+        if (error instanceof CustomError) {
+            super(error.message);
```
Comment (akshayvijayjain) [inline checkr/src/common/interfaces/custom-error.ts:6]: I feel, you can throw from constructor, if error is already instance of CustomError.

thus we avoid created nested CustomError.

yes, this is what I was referring in my previous comment

**Review by akshayvijayjain — APPROVED** [review]

**Pull request merged**